File: SelectionSort/SelectionSort.py
def ObetenerNumerosArchivo(ruta:str):
    numeros = []
    with open(ruta,"r") as f:
        for line in f:
            numeros.append(int(line))
    return numeros

def SelectionSort(arr:list,n:int):
    for i in range(0,n):
        j = i + 1
        key = arr[i]
        while(j<n):
            min = arr[j]
            if arr[i] > arr[j]:
                min = arr[j]
            j+=1
        arr[i] = min
        arr[j-1] = key

def GenerarArchivo(numeros,n):
    with open("./numeros_ordenados.txt","w") as f:
        for i in range(0,n):
            f.write(str(numeros[i]))
            if i != n-1:
                f.write("\n")

def run():
    #numeros = ObetenerNumerosArchivo("numeros.txt")
    A = [14,33,27,10,25,35,19,23,42,44]
    n = len(A)
    SelectionSort(A,n)
    print(A)
    #GenerarArchivo(numeros,len(numeros))
if __name__ == '__main__':
    run()

File: SelectionSort/SelectionSort.c
#include <stdio.h>
#include <stdlib.h>

void SelectionSort(int* arr,int n);
int main(){ 
    int A[] = {14,33,27,10,35,19,42,44};
    int n = 8;
    for (int i=0;i<n;i++){
        printf("%d ",A[i]);
    }
    SelectionSort(A,n);

    printf("\n");
    for (int i=0;i<n;i++){
        printf("%d ",A[i]);
    }
}

void SelectionSort(int* arr,int n){
    int i,j,min;
    for(i=0;i<n;i++){
        j = i + 1;
        while(j<n){
            if(arr[j] < arr[i]){
                min = arr[i];
                arr[i] = arr[j];
                arr[j] = min;
            }
            j+=1;
        }
    }
}